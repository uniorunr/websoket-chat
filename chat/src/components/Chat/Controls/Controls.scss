@import "../../../theme/colors";

.status {
  width: 100%;
  display: flex;
  justify-content: center;
  border: 2px solid $black-color;
  border-radius: 5px 5px 0 0;
  font-size: 1.4rem;
  background: $white-color;

  .status__content {
    width: 100%;
    padding: 10px 20px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .status__wrapper {
      flex-grow: 1;
      display: flex;
      align-items: center;

      .status__status {
        font-weight: 700;
        margin: 5px 0;

        .status__value {
          text-transform: uppercase;
        }

        .status__value_offline {
          color: $error-color;
        }

        .status__value_online {
          color: $success-color;
        }
      }

      .status__status-mobile_online {
        display: none;
        background: $success-color;
        width: 15px;
        height: 15px;
        border-radius: 50%;
      }

      .status__status-mobile_offline {
        display: none;
        background: $error-color;
        width: 15px;
        height: 15px;
        border-radius: 50%;
      }
    }

    .user-name {
      margin: 5px 20px;
      font-size: 1.4rem;
      font-weight: 700;
      color: $black-color;
    }
  }
}

@media only screen and (max-width : 549px) {
  .status {
    .status__content {
      .status__wrapper {
        .status__status {
          display: none;
        }

        .status__status-mobile {
          display: block;
        }
      }
    }
  }
}

