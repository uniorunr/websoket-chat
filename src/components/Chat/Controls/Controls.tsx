import React from 'react';
import { connect } from 'react-redux';
import { StoreState } from '../../../types';
import LogoutButton from './LogoutButton/LogoutButton';
import ThemeControl from './ThemeContrrol/ThemeControl';
import './Controls.scss';

const Controls = ({ chatStatus, userName }: StatusProps) => {
  return (
    <div className="controls">
      <div className="controls__wrapper">
        <div className="status">
          <div
            className={`status__status-mobile status__status-mobile_${chatStatus}`}
          />
          <p className="status__status">
            Status:{' '}
            <span className={`status__value status__value_${chatStatus}`}>
              {chatStatus}
            </span>
          </p>
        </div>
        <ThemeControl />
        <span className="user-name">{userName}</span>
        <LogoutButton />
      </div>
    </div>
  );
};

interface StatusProps {
  chatStatus: string;
  userName: string;
}

const mapStateToProps = (state: StoreState) => ({
  chatStatus: state.chatStatus,
  userName: state.userName
});

export default connect(mapStateToProps)(Controls);
