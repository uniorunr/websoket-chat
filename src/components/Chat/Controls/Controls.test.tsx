import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/mainStore';
import Controls from './Controls';

it('Controls component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Controls />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
