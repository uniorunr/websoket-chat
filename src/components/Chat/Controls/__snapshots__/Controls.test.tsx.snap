// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders correctly 1`] = `
<div
  className="controls"
>
  <div
    className="controls__wrapper"
  >
    <div
      className="status"
    >
      <div
        className="status__status-mobile status__status-mobile_offline"
      />
      <p
        className="status__status"
      >
        Status:
         
        <span
          className="status__value status__value_offline"
        >
          offline
        </span>
      </p>
    </div>
    <div
      className="theme-control"
    >
      <button
        className="theme-control__button-mobile"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Theme choose icon"
          src="rainbow.png"
          width={25}
        />
      </button>
      <button
        className="theme-control__button"
        onClick={[Function]}
        type="button"
      >
        settings
      </button>
    </div>
    <span
      className="user-name"
    >
      
    </span>
    <div
      className="logout"
    >
      <button
        className="logout__button-mobile"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Logout Icon"
          src="logout.png"
          width={20}
        />
      </button>
      <button
        className="logout__button"
        onClick={[Function]}
        type="button"
      >
        logout
      </button>
    </div>
  </div>
</div>
`;
